examples/oneapi/cpp/source/graph_service_functions/graph_service_functions.cpp

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include <iostream>
+
+#include "oneapi/dal/data/undirected_adjacency_array_graph.hpp"
+#include "oneapi/dal/data/graph_service_functions.hpp"
+#include "oneapi/dal/util/csv_data_source.hpp"
+#include "oneapi/dal/util/load_graph.hpp"
+
+using namespace oneapi::dal;
+using namespace oneapi::dal::preview;
+
+int main(int argc, char **argv) {
+    if (argc < 2) return 0;
+    std::string filename = argv[1];
+    csv_data_source ds(filename);
+    edge_list_to_csr_descriptor d;
+    auto my_graph = load_graph<graph>(d, ds);
+	std::cout << "Number of vertices: " << get_vertex_count(my_graph) << std::endl;
+	std::cout << "Number of edges: " <<get_edge_count(my_graph) << std::endl;
+	auto node_id = 0;
+	std::cout << "Degree of " << node_id << ": "
+			<< get_vertex_degree(my_graph, node_id) << std::endl;
+	for (unsigned int j = 0; j < get_vertex_count(my_graph); ++j) {
+		std::cout << "Neighbors of " << j << ": ";
+		auto neigh = get_vertex_neighbors(my_graph, j);
+		for (auto i = neigh.first; i != neigh.second; ++i){
```

Comment:
is this pair?
auto hides the type
can we do auto i : ...
Style guide question on the variable names in the example